controllers/useraccount/console_usersettings.go

```diff
@@ -0,0 +1,204 @@
+			return err
+		} else if len(configMapList) > 0 {
+			// if the number of items retrieved are more than one, should delete all of them
+			for key := range configMapList {
+				toDelete = append(toDelete, &configMapList[key])
+			}
+		}
+	} else if configMap != nil {
+		toDelete = append(toDelete, configMap)
+	} else {
+		// there was some kind of error in getting
+		return err
+	}
+	if err := deleteResources(ctx, cl, toDelete); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteRole deletes a Role associated with a user from console setting.
+// It first attempts to find the Role by name. If not found, it then looks for the Role by label.
+// If multiple Roles are found by label, all of them are deleted.
+// The function returns an error if any occurred.
+func deleteRole(ctx context.Context, cl client.Client, userUID string) error {
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting role with name %s", name))
+
+	var toDelete []client.Object
+	objectType := &v1.Role{TypeMeta: metav1.TypeMeta{Kind: "Role"}}
+
+	role, err := getConsoleUserSettingObjectByName(ctx, cl, name, objectType)
+	if err == nil && role == nil {
+		// could not find the role by name, try finding the role by label
+		if roleList, err := getRolesByLabel(ctx, cl, userUID); err != nil || len(roleList) == 0 {
+			return err
+		} else if len(roleList) > 0 {
+			for key := range roleList {
+				toDelete = append(toDelete, &roleList[key])
+			}
+		}
+	} else if role != nil {
+		toDelete = append(toDelete, role)
+	} else {
+		// there was some kind of error in getting
+		return err
+	}
+	if err := deleteResources(ctx, cl, toDelete); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteRoleBinding deletes a Rolebinding associated with a user from console setting.
+// It first attempts to find the Rolebinding by name. If not found, it then looks for the Rolebinding by label.
+// If multiple Rolebindings are found by label, all of them are deleted.
+// The function returns an error if any occurred.
+func deleteRoleBinding(ctx context.Context, cl client.Client, userUID string) error {
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting rolebinding with name %s", name))
+
+	var toDelete []client.Object
+	objectType := &v1.RoleBinding{TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"}}
+
+	rb, err := getConsoleUserSettingObjectByName(ctx, cl, name, objectType)
+	if err == nil && rb == nil {
+		// try with label
+		if rbList, err := getRoleBindingsByLabel(ctx, cl, userUID); err != nil || len(rbList) == 0 {
+			return err
+		} else if len(rbList) > 0 {
+			for key := range rbList {
+				toDelete = append(toDelete, &rbList[key])
+			}
+		}
+	} else if rb != nil {
+		toDelete = append(toDelete, rb)
+	} else {
+		// there was some kind of error in getting
+		return err
+	}
```

Comment:
same here

Reply:
thanks, simplified in [d4bc88c](https://github.com/codeready-toolchain/member-operator/pull/568/commits/d4bc88cc583bfad7f84185b5fa341ffc508984e3)